examples/seq2seq/seq2seq.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import collections
+
+from nltk.corpus import comtrans
+from nltk.translate import bleu_score
+import numpy
+
+import chainer
+from chainer import cuda
+import chainer.functions as F
+import chainer.links as L
+from chainer import reporter
+from chainer import training
+from chainer.training import extensions
+
+
+def sequence_embed(embed, xs):
+    x_len = [len(x) for x in xs]
+    x_section = numpy.cumsum(x_len[:-1])
+    ex = embed(F.concat(xs, axis=0))
+    exs = F.split_axis(ex, x_section, 0, force_tuple=True)
+    return exs
+
+
+class Seq2seq(chainer.Chain):
+
+    def __init__(self, n_layers, n_source_vocab, n_target_vocab, n_units):
+        super(Seq2seq, self).__init__(
+            embed_x=L.EmbedID(n_source_vocab, n_units),
+            embed_y=L.EmbedID(n_target_vocab, n_units),
+            encoder=L.NStepLSTM(n_layers, n_units, n_units, 0.1),
+            decoder=L.NStepLSTM(n_layers, n_units, n_units, 0.1),
+            W=L.Linear(n_units, n_target_vocab),
+        )
+        self.n_layers = n_layers
+        self.n_units = n_units
+
+    def __call__(self, *inputs):
+        xs = inputs[:len(inputs) // 2]
+        ys = inputs[len(inputs) // 2:]
+
+        eos = self.xp.array([0], 'i')
+        ys_in = [F.concat([eos, y], axis=0) for y in ys]
+        ys_out = [F.concat([y, eos], axis=0) for y in ys]
+
+        exs = sequence_embed(self.embed_x, xs)
+        eys = sequence_embed(self.embed_y, ys_in)
+
+        batch = len(xs)
+        # Initial hidden variable and cell variable
+        zero = self.xp.zeros((self.n_layers, batch, self.n_units), 'f')
+        hx, cx, _ = self.encoder(zero, zero, exs)
+        _, _, os = self.decoder(hx, cx, eys)
+        loss = F.softmax_cross_entropy(
+            self.W(F.concat(os, axis=0)), F.concat(ys_out, axis=0))
```

Comment:
You simply ignore the variation of normalization coefficients to devide in `softmax_cross_entropy` for each batch, don't you? A typical coefficient is the batchsize (= `len(ys_out)`), while this implementation uses the total numbers of whole tokens in a batch (= `sum(y.shape[0] for y in ys_out)`) for it.
(If you already recognized it and do so for simplicity, that's fine.)